src/transxchange/naptan.rs

```diff
@@ -0,0 +1,264 @@
+    fn coord(&self) -> Result<Coord> {
+        let epsg_27700_point = Point::new(self.easting, self.northing);
+        let converter = Proj::new_known_crs(&EPSG_27700, &EPSG_4326, None).ok_or_else(|| {
+            format_err!(
+                "Proj cannot build a converter from '{}' to '{}'",
+                EPSG_27700,
+                EPSG_4326
+            )
+        })?;
+        converter.convert(epsg_27700_point).map(Coord::from)
+    }
+}
+
+fn read_stop_areas<R>(reader: R) -> Result<CollectionWithId<NTMStopArea>>
+where
+    R: Read,
+{
+    let mut reader = csv::ReaderBuilder::new()
+        .delimiter(b',')
+        .trim(csv::Trim::All)
+        .from_reader(reader);
+    for record in reader.deserialize() {
+        let _stop_area: StopArea =
+            record.with_context(|_| "Error parsing the CSV record into a StopArea")?;
+        // TODO: Load stop_area into a collection
+    }
+    unimplemented!()
+}
+
+fn read_stops_in_area<R>(reader: R) -> Result<HashMap<String, String>>
+where
+    R: Read,
+{
+    let mut reader = csv::ReaderBuilder::new()
+        .delimiter(b',')
+        .trim(csv::Trim::All)
+        .from_reader(reader);
+    for record in reader.deserialize() {
+        let _stop_in_area: StopInArea =
+            record.with_context(|_| "Error parsing the CSV record into a StopInArea")?;
+        // TODO: Load relations between Stops and Stop Areas in a map where the
+        //       key is the Stop ID and the value is the Stop Area ID
+    }
+    unimplemented!()
+}
+
+fn read_stops<R>(
+    reader: R,
+    _stops_in_area: &HashMap<String, String>,
+) -> Result<CollectionWithId<NTMStopPoint>>
+where
+    R: Read,
+{
+    let mut reader = csv::ReaderBuilder::new()
+        .delimiter(b',')
+        .trim(csv::Trim::All)
+        .from_reader(reader);
+    for record in reader.deserialize() {
+        let _stop: Stop = record.with_context(|_| "Error parsing the CSV record into a Stop")?;
+    }
+    unimplemented!()
+}
+
+fn validate_stops(
+    _stop_areas: &CollectionWithId<NTMStopArea>,
+    _stop_points: &CollectionWithId<NTMStopPoint>,
+) -> Result<()> {
+    unimplemented!()
+}
+
+pub fn read_naptan<P>(naptan_path: P, collections: &mut Collections) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    #[derive(Clone, Copy, Eq, Hash, PartialEq)]
+    enum FileType {
+        Stops,
+        StopAreas,
+        StopsInArea,
+    };
+    impl std::fmt::Display for FileType {
+        fn fmt(&self, f: &mut std::fmt::Formatter) -> std::result::Result<(), std::fmt::Error> {
+            match self {
+                FileType::StopAreas => write!(f, "StopAreas"),
+                FileType::StopsInArea => write!(f, "StopsInArea"),
+                FileType::Stops => write!(f, "Stops"),
+            }
+        }
+    }
+    let zip_file = File::open(naptan_path)?;
+    let mut zip_archive = ZipArchive::new(zip_file)?;
+    let mut files = HashMap::new();
+    for index in 0..zip_archive.len() {
```

Comment:
you can't use http://mvdnes.github.io/rust-docs/zip-rs/zip/read/struct.ZipArchive.html#method.by_name ?

Reply:
Let's consider my solution a dumb one and try yours instead :stuck_out_tongue_closed_eyes:

Reply:
Much simpler (like by a lot). Thanks (see 7b7fc368a7bd98b432f821d641f6e1186249e4b6).